### DISCUSSION TITLE
The following material properties are declared on block 2 by multiple materials

### DISCUSSION THREAD

**Discussion by Rahim-Habibi** [idaholab/moose#21477] [Q&A Modules: Solid mechanics] [answered]
Hi, 

I have ran a Thermo-Hydro-Mechanical response of a reservoir against fluid injection successfully. Now, in the next step, I trying to add failure criterion like Mohr-Coulomb to my model. I have got knoweldge how to apply Mohr-Coulomb from some examples such as:
https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/test/tests/multi/paper3.i

I just ran this example succesfully, but when I run the model the following errors pops up: 
The following material properties are declared on block 2 by multiple materials
![image](https://user-images.githubusercontent.com/57484835/176856177-831304c5-4512-4106-a68a-82ea34936207.png)

By the way, I used Thermal, Hydraulic and Mechanical material objects in the model, a part of it as follows:
```
[Materials]
  [./mc]
    type = ComputeMultiPlasticityStress
    ep_plastic_tolerance = 1e-3
    plastic_models = 'tensile mc'
    max_NR_iterations = 5
    specialIC = 'rock'
    debug_fspb = crash
    block = '2 3 4 11 12 13 14 15 16 17 18 19 20 21'
  [../]
  [./strain]
    type = ComputeFiniteStrain
    displacements = 'disp_x disp_y disp_z'
    block = '2 3 4 11 12 13 14 15 16 17 18 19 20 21'
  [../]
  [./Elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 15e9
    poissons_ratio = 0.2
    block = '2 3 4 11 12 13 14 15 16 17 18 19 20 21'
  [../]
  [./thermal_expansion]
    type = ComputeThermalExpansionEigenstrain
    thermal_expansion_coeff = 1e-5
    temperature = temperature
    stress_free_temperature = temperature
    eigenstrain_name = 'thermal_eigenstrain'
    block = '2 3 4 11 12 13 14 15 16 17 18 19 20 21'
  [../]
```
It seems that there is kind of confliction between the objects declaring the material. 
Does anyone know how can I handle this issue?

Thanks in advance 
Rahim

**Comment by GiudGiud** [comment, marked as answer]
Hi Rahim

Do you have an object named`all_strain` ?
Are you using the tensor mechanics action (Modules/TensorMechanics syntax) as well ? 
Could you please attach your input file?

Format-wise:
- please do not post screenshots of text, they are not searchable
- please format code and input file snippets using triple quotes

Guillaume

**Comment by RWTHLHK** [comment]
Hello, do you fix your problem? I met a similar one.

**Reply by Rahim-Habibi** [reply]
Hi, 
Yes, I fixed it. it means that you declared the same property in two blocks in the input file. In my case, strain was declared in two blocks. 
If you attach your input file, I can help you more. 
Rahim

**Reply by RWTHLHK** [reply]
Ah, thanks for reminding me. I just fixed the problem, too.
